Fixes:  #1117, #1146 RecentlyPlayed - Tablet (Landscape & Portrait) (Low-fi)

MohamedMedhat1998 opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
- Implemented low-fi UI for RecentlyPlayed fragment landscape & portrait mode
- Fixes #1117 
- Fixes #1146 
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

MohamedMedhat1998 committed:
-Added some xml files for tablet-layout (sw600dp-land)
-Added a test class for testing "SpanCount" in different configurations

MohamedMedhat1998 committed:
-Fixed " Layout state should be one of 100 but it is 10" error
-Another error appeared "KotlinNullPointerException"

MohamedMedhat1998 committed:
-Fixed "kotlin.KotlinNullPointerException"
-Added "RecentlyPlayedFragmentActivityTest" and its "presenter"

MohamedMedhat1998 committed:
Merge remote-tracking branch 'origin/develop' into recently-played-tablet-land-lowfi-2

MohamedMedhat1998 committed:
-Updated the "RecentlyPlayedSpanTest" class

MohamedMedhat1998 committed:
Merge remote-tracking branch 'origin/develop' into recently-played-tablet-land-lowfi-2

MohamedMedhat1998 committed:
-Implemented RecentlyPlayed Landscape & Portrait low-fi UI

MohamedMedhat1998 committed:
-Moved the tag of "RecentlyPlayedFragment" to its companion object

rt4914 commented on app/src/main/java/org/oppia/app/home/recentlyplayed/RecentlyPlayedActivityPresenter.kt:
```diff
@@ -13,7 +13,8 @@ class RecentlyPlayedActivityPresenter @Inject constructor(private val activity:
         R.id.recently_played_fragment_placeholder,
-        RecentlyPlayedFragment.newInstance(internalProfileId)
+        RecentlyPlayedFragment.newInstance(internalProfileId),
+        RecentlyPlayedFragment.TAG_RECENTLY_PLAYED_FRAGMENT
```
Nice. Thanks for doing this.

MohamedMedhat1998 replied:
Welcome!

rt4914 commented on app/src/main/java/org/oppia/app/home/recentlyplayed/RecentlyPlayedFragment.kt:
```diff
@@ -14,6 +14,7 @@ private const val RECENTLY_PLAYED_FRAGMENT_INTERNAL_PROFILE_ID_KEY = "RecentlyPl
 /** Fragment that contains all recently played stories. */
 class RecentlyPlayedFragment : InjectableFragment(), OngoingStoryClickListener {
   companion object {
+    const val TAG_RECENTLY_PLAYED_FRAGMENT = "recently-played-fragment"
```
The value should be `TAG_RECENTLY_PLAYED_FRAGMENT`

MohamedMedhat1998 replied:
Done!

rt4914 commented on app/src/main/res/layout/recently_played_fragment.xml:
```diff
@@ -47,7 +47,8 @@
         android:paddingBottom="172dp"
-        android:scrollbars="none" />
+        android:scrollbars="none"
+        android:tag="recently_played_recycler_view" />
```
Why do we need this tag?

MohamedMedhat1998 replied:
It was added in a middle change that was canceled later.  
That is adding robolectric tests for the span count for each configuration, but I have modified the test cases themselves instead.  
I've removed the tag. PTAL

rt4914 replied:
OKay. Thanks.

rt4914 commented on app/src/sharedTest/java/org/oppia/app/home/RecentlyPlayedFragmentTest.kt:
```diff
@@ -530,7 +534,11 @@ class RecentlyPlayedFragmentTest {
         )
       )
-      onView(withId(R.id.ongoing_story_recycler_view)).check(hasGridItemCount(2, 2))
+      if (context.resources.getBoolean(R.bool.isTablet)) {
```
Update the test case name:
`testRecentlyPlayedTestActivity_checkSpanForItem2_spanSizeIsTwo OrThree` 

Ditto for above test case.

MohamedMedhat1998 replied:
Done!

rt4914 reviewed (approved):
LGTM, Suggested test case renaming.

MohamedMedhat1998 committed:
-Updated the "RecentlyPlayedFragment" tag value
-Removed the unnecessary tag in the "RecyclerView" in "recently_played_fragment" layout
-Renamed some test cases in "RecentlyPlayedFragmentTest"
